fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -200,10 +200,20 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
                 override fun afterTextChanged(editable: Editable?) {
                     // update usernameChanged to true when the text is not empty,
                     // otherwise it is not changed, as this screen starts with an empty username.
-                    usernameChanged = editable.toString().isNotEmpty()
-                    updateUsernameField()
-                    setSaveButtonState()
-                    findDuplicate()
+                    when {
+                        editable.toString().isEmpty() -> {
+                            validUsername = false
+                            binding.clearUsernameTextButton.isVisible = false
```

Comment:
Based on the model of the password field this is where we should also call `updateUsernameField` to set an error for empty username.
(`usernameChanged` should probably be changed to `true` here also)